myst_nb/core/utils.py

```diff
@@ -25,9 +25,12 @@ def coalesce_streams(outputs: list[NotebookNode]) -> list[NotebookNode]:
     for output in outputs:
         if output["output_type"] == "stream":
             if output["name"] in streams:
-                streams[output["name"]]["text"] += output["text"]
+                out = output["text"].strip()
+                if out:
+                    streams[output["name"]]["text"] += f"{out}\n"
             else:
                 new_outputs.append(output)
+                output["text"] = output["text"].strip() + "\n"
```

Comment:
Is this line necessary?

Reply:
Yes, perhaps it would be clearer to move it one line up. We're mutating the dict we just added to `new_outputs`.

Reply:
oh, I was just nitpicking that you stripe off whitespace and add a newline; practically the same, so is this really needed?

Reply:
Oh this is in case there are multiple newlines at the end which can happen after merging the cells.

See the example in my first post.

I am hard at work on trying to write a test. Turns out that on MacOS the issue does not exist ... took me a good while to realize that.

Reply:
@bsipocz, just added a test!

See my comment here https://github.com/executablebooks/MyST-NB/pull/604#issuecomment-2362274297

Reply:
Thanks! I'll open a follow-up to fix that.

Reply:
Good point! I just opened https://github.com/executablebooks/MyST-NB/pull/640